exec/java-exec/src/main/java/org/apache/drill/exec/store/log/LogRecordReader.java

```diff
@@ -0,0 +1,750 @@
+import java.util.regex.Pattern;
+import java.util.regex.PatternSyntaxException;
+
+public class LogRecordReader extends AbstractRecordReader {
+
+  private static final org.slf4j.Logger logger = org.slf4j.LoggerFactory.getLogger(LogRecordReader.class);
+
+  private abstract static class ColumnDefn {
+    public final String name;
+    public final int index;
+    public final String format;
+
+    public ColumnDefn(String name, int index) {
+      this.name = name;
+      this.index = index;
+      this.format = "";
+    }
+
+    public ColumnDefn(String name, int index, String format) {
+      this.name = name;
+      this.index = index;
+      this.format = format;
+    }
+
+    public abstract void define(OutputMutator outputMutator) throws SchemaChangeException;
+
+    public abstract void load(int rowIndex, String value);
+
+    @Override
+    //For testing
+    public String toString() {
+      return "Name: " + name + ", Index: " + index;
+    }
+  }
+
+  private static class VarCharDefn extends ColumnDefn {
+
+    private NullableVarCharVector.Mutator mutator;
+
+    public VarCharDefn(String name, int index) {
+      super(name, index);
+    }
+
+    @Override
+    public void define(OutputMutator outputMutator) throws SchemaChangeException {
+      MaterializedField field = MaterializedField.create(name,
+          Types.optional(MinorType.VARCHAR));
+      mutator = outputMutator.addField(field, NullableVarCharVector.class).getMutator();
+    }
+
+    @Override
+    public void load(int rowIndex, String value) {
+      mutator.set(rowIndex, value.getBytes());
+    }
+  }
+
+  private static class BigIntDefn extends ColumnDefn {
+
+    private NullableBigIntVector.Mutator mutator;
+
+    public BigIntDefn(String name, int index) {
+      super(name, index);
+    }
+
+    @Override
+    public void define(OutputMutator outputMutator) throws SchemaChangeException {
+      MaterializedField field = MaterializedField.create(name,
+          Types.optional(MinorType.BIGINT));
+      mutator = outputMutator.addField(field, NullableBigIntVector.class).getMutator();
+    }
+
+    @Override
+    public void load(int rowIndex, String value) {
+      try {
+        mutator.set(rowIndex, Long.parseLong(value));
+      } catch (NumberFormatException e) {
+        throw UserException
+            .dataReadError(e)
+            .addContext("Failed to parse an INT field")
+            .addContext("Column", name)
+            .addContext("Position", index)
+            .addContext("Value", value)
+            .build(logger);
+      }
+    }
+  }
+
+  private static class SmallIntDefn extends ColumnDefn {
+
+    private NullableSmallIntVector.Mutator mutator;
+
+    public SmallIntDefn(String name, int index) {
+      super(name, index);
+    }
+
+    @Override
+    public void define(OutputMutator outputMutator) throws SchemaChangeException {
+      MaterializedField field = MaterializedField.create(name,
+          Types.optional(MinorType.BIGINT));
```

Comment:
Hmm... Kind of confused. The class name is `SmallIntDefn`, the mutter is for `NullableSmallIntVector`, but the minor type being used is `BIGINT`.

In fact, this tells us that we never tested this field type. Should we go back and make sure that we have a consistent story across accepted type names, supported types internally, and unit tests?